internal/executor/util/pod_util.go

```diff
@@ -125,3 +128,52 @@ func MergePodList(list1 []*v1.Pod, list2 []*v1.Pod) []*v1.Pod {
 
 	return allPods
 }
+
+func FilterPods(pods []*v1.Pod, filter func(*v1.Pod) bool) []*v1.Pod {
+	result := make([]*v1.Pod, 0)
+	for _, pod := range pods {
+		if filter(pod) {
+			result = append(result, pod)
+		}
+	}
+	return result
+}
+
+func LastStatusChange(pod *v1.Pod) (time.Time, error) {
+	maxStatusChange := pod.CreationTimestamp.Time
+	for _, containerStatus := range pod.Status.ContainerStatuses {
+		if s := containerStatus.State.Running; s != nil {
+			maxStatusChange = maxTime(maxStatusChange, s.StartedAt.Time)
+		}
+		if s := containerStatus.State.Terminated; s != nil {
+			maxStatusChange = maxTime(maxStatusChange, s.FinishedAt.Time)
+		}
+	}
+
+	for _, condition := range pod.Status.Conditions {
+		maxStatusChange = maxTime(maxStatusChange, condition.LastTransitionTime.Time)
+	}
+
+	if maxStatusChange.IsZero() {
+		return maxStatusChange, errors.New("cannot determine last status change")
+	}
+	return maxStatusChange, nil
+}
+
+func FindLastContainerStartTime(pod *v1.Pod) time.Time {
```

Comment:
This func doesn't really do what it says it does, at least not entirely. Also it overlaps nicely with the above func. Maybe we could improve naming or something here.